plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/TrinoDeleteFile.java

```diff
@@ -0,0 +1,317 @@
+                icebergDeleteFile.keyMetadata() == null ? null : icebergDeleteFile.keyMetadata().array(),
+                icebergDeleteFile.equalityFieldIds(),
+                icebergDeleteFile.sortOrderId(),
+                icebergDeleteFile.splitOffsets());
+    }
+
+    @JsonCreator
+    public TrinoDeleteFile(
+            @JsonProperty("pos") Long pos,
+            @JsonProperty("specId") int specId,
+            @JsonProperty("fileContent") FileContent fileContent,
+            @JsonProperty("path") CharSequence path,
+            @JsonProperty("format") FileFormat format,
+            @JsonProperty("recordCount") long recordCount,
+            @JsonProperty("fileSizeInBytes") long fileSizeInBytes,
+            @JsonProperty("columnSizes") Map<Integer, Long> columnSizes,
+            @JsonProperty("valueCounts") Map<Integer, Long> valueCounts,
+            @JsonProperty("nullValueCounts") Map<Integer, Long> nullValueCounts,
+            @JsonProperty("nanValueCounts") Map<Integer, Long> nanValueCounts,
+            @JsonProperty("lowerBounds") Map<Integer, byte[]> lowerBounds,
+            @JsonProperty("upperBounds") Map<Integer, byte[]> upperBounds,
+            @JsonProperty("keyMetadata") byte[] keyMetadata,
+            @JsonProperty("equalityFieldIds") List<Integer> equalityFieldIds,
+            @JsonProperty("sortOrderId") Integer sortOrderId,
+            @JsonProperty("splitOffsets") List<Long> splitOffsets)
+    {
+        this.pos = pos;
+        this.specId = specId;
+        this.fileContent = fileContent;
+        this.path = path;
+        this.format = format;
+        this.recordCount = recordCount;
+        this.fileSizeInBytes = fileSizeInBytes;
+        this.columnSizes = columnSizes;
+        this.valueCounts = valueCounts;
+        this.nullValueCounts = nullValueCounts;
+        this.nanValueCounts = nanValueCounts;
+        this.lowerBounds = lowerBounds;
+        this.upperBounds = upperBounds;
+        this.keyMetadata = keyMetadata;
+        this.equalityFieldIds = equalityFieldIds;
+        this.sortOrderId = sortOrderId;
+        this.splitOffsets = splitOffsets;
+    }
+
+    @Override
+    @JsonProperty("pos")
+    public Long pos()
+    {
+        return pos;
+    }
+
+    @Override
+    @JsonProperty("specId")
+    public int specId()
+    {
+        return specId;
+    }
+
+    @Override
+    @JsonProperty("fileContent")
+    public FileContent content()
+    {
+        return fileContent;
+    }
+
+    @Override
+    @JsonProperty("path")
+    public CharSequence path()
+    {
+        return path;
+    }
+
+    @Override
+    @JsonProperty("format")
+    public FileFormat format()
+    {
+        return format;
+    }
+
+    // TODO: Probably need to figure out how to serialize this
+    @Override
+    public StructLike partition()
+    {
+        throw new IllegalArgumentException("Cannot call 'partition' method on TrinoDeleteFile");
```

Comment:
If this is not needed, kist add `@JsonIgnore` + `throw new UnsupportedOperationException()`